src/components/Sidebaritem/index.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-script-url, react/prop-types */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Consumer as NavigationConsumer } from '../Sidebar/context';
+import RenderIf from '../RenderIf';
+import './styles.css';
+
+function Item(props) {
+    const {
+        label,
+        icon,
+        name,
+        href,
+        onClick,
+        className,
+        style,
+        selectedItem,
+        onSelect,
+        activeColor,
+    } = props;
+    const isSelected = name === selectedItem;
+    const activeLineStyles = { backgroundColor: activeColor };
+
+    const getContainerClassNames = () => classnames('rainbow-sidebar-item', {
+        'rainbow-sidebar-item--active': isSelected,
+    }, className);
+
+    const getAriaCurrent = () => {
+        if (isSelected) {
+            return 'page';
+        }
+        return undefined;
+    };
+
+    function hanldeOnClick(e) {
+        onClick(e);
+        onSelect(e, name);
+    }
+
+    return (
+        <li className={getContainerClassNames()} style={style}>
+            <RenderIf isTrue={isSelected}>
+                <div className="rainbow-sidebar-item-line--active" style={activeLineStyles} />
+            </RenderIf>
+            <a
+                href={href}
+                onClick={hanldeOnClick}
+                className="rainbow-sidebar-item_action"
+                aria-current={getAriaCurrent()}>
+                <div className="rainbow-sidebar-item_icon" >{icon}</div>
+                <RenderIf isTrue={!!label}>
+                    <span className="rainbow-sidebar-item_label" >{label}</span>
+                </RenderIf>
+            </a>
+        </li>
+    );
+}
+
+/**
+ * @category Layout
+ */
+export default function SidebarItem(props) {
+    return (
+        <NavigationConsumer>
+            {context => (
+                <Item
+                    {...props}
+                    {...context} />
+            )}
+        </NavigationConsumer>
+    );
+}
```

Comment:
@LeandroTorresSicilia is there a way to use the consumer as a high order function instead of using declarative component ??

Reply:
this is the new context API, we could just wrap it in a high order component and export the component through the HOC